packages/service-library/src/servicelib/decorators.py

```diff
@@ -20,16 +24,29 @@ def safe_return(if_fails_return=False, catch=None, logger=None):
 
     def decorate(func):
         @wraps(func)
-        def safe_func(*args, **kargs):
+        def safe_func(*args, **kwargs):
             try:
-                res = func(*args, **kargs)
-                return res
+                return func(*args, **kwargs)
             except catch as err:
                 logger.info("%s failed:  %s", func.__name__, str(err))
             except Exception:  # pylint: disable=broad-except
                 logger.info("%s failed unexpectedly", func.__name__, exc_info=True)
-            return deepcopy(if_fails_return)  # avoid issues with default mutables
+            return deepcopy(if_fails_return)  # avoid issues with default mutable
 
         return safe_func
 
     return decorate
+
+
+def async_delayed(
+    interval: datetime.timedelta,
+) -> Callable[..., Callable[..., Coroutine]]:
```

Comment:
Q: must this be a Coroutine? not Awaitable[Any] ?

Reply:
Apparently this plays nice with `asyncio.create_task(` which expects a coroutine. also it is correct, `wrapper` is a coroutine. So I think it's OK like this.